File: JAVA/3016. Minimum Number of Pushes to Type Word II.java
class Solution {
    public int minimumPushes(String word) {
        int[] mp = new int[26];
        for (char ch : word.toCharArray()) {
            mp[ch - 'a']++;
        }
        Integer[] mpInteger = Arrays.stream(mp).boxed().toArray(Integer[]::new);
        Arrays.sort(mpInteger, Comparator.reverseOrder());
        int ans = 0;
        int weight = 1;
        for (int i = 0; i < 26 && mpInteger[i] > 0; i++) {
            ans += mpInteger[i] * weight;
            if ((i + 1) % 8 == 0) {
                weight++;
            }
        }
        return ans;
    }
}

File: C++/3016. Minimum Number of Pushes to Type Word II.cpp
class Solution {
public:
    int minimumPushes(string word) {
        unordered_map<char,int>m;
        for(auto it : word)
        {
            m[it]++;
        }
        vector<int>v;
        for(auto it : m)
        {
            v.push_back(it.second);
        }
        sort(v.begin(),v.end(),greater<>());
        int place = 8;
        int press = 1;
        int ans = 0;
        for(auto it : v){
            if(place == 0)
            {
                press++;
                place = 8;
            }
            ans += it*press;
            place--;
        }
        return ans;        
    }
};
